activiti-cloud-services-runtime-bundle/activiti-cloud-services-job-executor/src/main/java/org/activiti/cloud/services/job/executor/JobMessageHandler.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.cloud.services.job.executor;
+
+import org.activiti.engine.ActivitiException;
+import org.activiti.engine.impl.asyncexecutor.ExecuteAsyncRunnable;
+import org.activiti.engine.impl.cfg.ProcessEngineConfigurationImpl;
+import org.activiti.engine.impl.interceptor.Command;
+import org.activiti.engine.impl.interceptor.CommandContext;
+import org.activiti.engine.impl.persistence.entity.JobEntity;
+import org.activiti.engine.runtime.Job;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.messaging.MessagingException;
+
+public class JobMessageHandler  implements MessageHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(JobMessageHandler.class);
+
+    private final ProcessEngineConfigurationImpl processEngineConfiguration;
+
+    public JobMessageHandler(ProcessEngineConfigurationImpl processEngineConfiguration) {
+        this.processEngineConfiguration = processEngineConfiguration;
+    }
+
+    @Override
+    public void handleMessage(Message<?> message) throws MessagingException {
+        try {
+            String jobId = new String((byte[]) message.getPayload());
+
+            logger.info("Received job message with id: " + jobId);
+            
+            // Let's try to find existing job by jobId
```

Comment:
The message handler execution is a single unit of work that fits into a single method nicely. It will only make sense to externalize `findJobById` and `executeJob` if we need to allow to override JobMessageHandler implementations...

Reply:
Well, when you create a method, it doesn't mean you should reuse it in several places or you need to override it. It can be used only to encapsulate some logic to simplify the caller method (and avoid inline comments). Inline comments are often considered as code smells (unless they are providing information that is not straightforward to get from the code, like this one https://github.com/Activiti/activiti-cloud-runtime-bundle-service/pull/188/files#diff-767a504c8251113456bd793dc386a1a9R52). This blog post covers that subject nicely, please have a look: https://adamtom.at/articles/comments-are-code-smells/.

Reply:
Fixed.